### PR TITLE
fix problem and add test

### PR DESCRIPTION
**Pull request by zhangzerui20**
#35

### CODE REVIEW

File: store/boltdbStore/routing.go [hunk 1]
```diff
@@ -18,53 +18,287 @@
...
 	"time"
 )
 
 type routingStore struct {
 	handler BoltHandler
 }
 
-// 新增一个路由配置
+const (
+	routingStoreType = "routing"
```
Comment (andrewshan) [inline store/boltdbStore/routing.go:34]: 统一改成tblNameRouting这种格式

File: store/boltdbStore/routing.go [hunk 2]
```diff
@@ -18,53 +18,287 @@
...
+	routingFieldOutBounds = "OutBounds"
+	routingFieldRevision = "Revision"
+	routingFieldModifyTime = "ModifyTime"
+)
+
+// CreateRoutingConfig Add a routing configuration
 func (r *routingStore) CreateRoutingConfig(conf *model.RoutingConfig) error {
-	//TODO
+	if conf.ID == "" || conf.Revision == "" {
+		log.Errorf("[Store][database] create routing config missing service id or revision")
```
Comment (andrewshan) [inline store/boltdbStore/routing.go:45]: [Store][boltdb]，其他一样

File: store/boltdbStore/routing.go [hunk 3]
```diff
@@ -18,53 +18,287 @@
...
+		return err
+	}
 	return nil
 }
 
-// 删除一个路由配置
+// DeleteRoutingConfig Delete a routing configuration
 func (r *routingStore) DeleteRoutingConfig(serviceID string) error {
-	//TODO
+	if serviceID == "" {
```
Comment (andrewshan) [inline store/boltdbStore/routing.go:88]: 这个应该是configID吧，不是serviceID

File: store/boltdbStore/routing.go [hunk 4]
```diff
@@ -18,53 +18,287 @@
...
+	if err != nil {
+		log.Errorf("get route config from kv error, %v", err)
+		return nil, err
+	}
+
+	return toRouteConf(routes), nil
 }
 
-// 根据服务名和命名空间拉取路由配置
+// GetRoutingConfigWithService Get routing configuration based on service name and namespace
```
Comment (andrewshan) [inline store/boltdbStore/routing.go:128]: 注释都用中文吧

File: store/boltdbStore/instance.go [hunk 5]
```diff
@@ -239,11 +261,11 @@ func (i *instanceStore) GetInstance(instanceID string) (*model.Instance, error)
 		log.Errorf("load instance from kv error, %v", err)
```
Comment (andrewshan) [inline store/boltdbStore/instance.go:261]: 这里要加上插件前缀

File: store/boltdbStore/routing.go (hunk 3 above)
Comment (zhangzerui20) [reply store/boltdbStore/routing.go:88]: > 这个应该是configID吧，不是serviceID
这里接口定义的就是 serviceId

File: store/boltdbStore/instance.go [hunk 6]
```diff
@@ -19,9 +19,12 @@ package boltdbStore
...
 	"errors"
+	"fmt"
+	api "github.com/polarismesh/polaris-server/common/api/v1"
 	"github.com/polarismesh/polaris-server/common/log"
 	"github.com/polarismesh/polaris-server/common/model"
 	"sort"
+	"strconv"
 	"strings"
 	"time"
 )
```
Comment (chenhengqi) [inline store/boltdbStore/instance.go:30]: Run gofmt/goimports before submitting.

File: store/boltdbStore/instance.go [hunk 7]
```diff
@@ -60,46 +69,48 @@ func (i *instanceStore) BatchAddInstances(instances []*model.Instance) error {
...
 }
 
-// 删除一个实例，实际是把valid置为false
+// DeleteInstance Delete an instance
 func (i *instanceStore) DeleteInstance(instanceID string) error {
 
-	if err := i.handler.DeleteValues(InstanceStoreType, []string{instanceID}); err != nil{
-		log.Errorf("update instance to kv error, %v", err)
+	if err := i.handler.DeleteValues(tblNameInstance, []string{instanceID}); err != nil{
+		log.Errorf("[Store][boltdb] update instance to kv error, %v", err)
```
Comment (chenhengqi) [inline store/boltdbStore/instance.go:106]: Delete instance ?

File: store/boltdbStore/instance.go [hunk 8]
```diff
@@ -112,101 +123,110 @@ func (i *instanceStore) BatchDeleteInstances(ids []interface{}) error {
 		realIds = append(realIds, id.(string))
 	}
 
-	if err := i.handler.DeleteValues(InstanceStoreType, realIds); err != nil{
-		log.Errorf("update instance to kv error, %v", err)
+	if err := i.handler.DeleteValues(tblNameInstance, realIds); err != nil{
```
Comment (chenhengqi) [inline store/boltdbStore/instance.go:126]: Rename all `Id` to `ID`, `Ids` to `IDs`

File: store/boltdbStore/instance.go [hunk 9]
```diff
@@ -112,101 +123,110 @@ func (i *instanceStore) BatchDeleteInstances(ids []interface{}) error {
...
+		log.Errorf("[Store][boltdb] update instance to kv error, %v", err)
```
Comment (chenhengqi) [inline store/boltdbStore/instance.go:127]: delete ?

File: store/boltdbStore/instance.go [hunk 10]
```diff
@@ -112,101 +123,110 @@ func (i *instanceStore) BatchDeleteInstances(ids []interface{}) error {
...
 	return nil
 }
 
-// 清空一个实例，真正删除
+// CleanInstance Delete an instance
 func (i *instanceStore) CleanInstance(instanceID string) error {
-	if err := i.handler.DeleteValues(InstanceStoreType, []string{instanceID}); err != nil{
-		log.Errorf("update instance to kv error, %v", err)
+	if err := i.handler.DeleteValues(tblNameInstance, []string{instanceID}); err != nil{
+		log.Errorf("[Store][boltdb] update instance to kv error, %v", err)
```
Comment (chenhengqi) [inline store/boltdbStore/instance.go:136]: update -> delete ?

File: store/boltdbStore/instance.go [hunk 11]
```diff
@@ -112,101 +123,110 @@ func (i *instanceStore) BatchDeleteInstances(ids []interface{}) error {
...
-	//	if ok {
-	//		ids[instance.ID()] = true
-	//	}
-	//})
-	//
-	//if err != nil {
-	//	log.Errorf("list instance in kv error, %v", err)
-	//	return nil, err
-	//}
+	err := i.handler.IterateFields(tblNameInstance, insFieldProto, &model.Instance{}, func(ins interface{}){
```
Comment (chenhengqi) [inline store/boltdbStore/instance.go:149]: This callback could be defined as a separate variable.

**Review by chenhengqi — CHANGES_REQUESTED** [review]

**Pull request merged**